gadget/gadget.go

```diff
@@ -388,8 +388,8 @@ func validateCrossVolumeStructure(structures []PositionedStructure, knownStructu
 				return fmt.Errorf("structure %v refers to an unknown structure %q",
 					ps, ps.OffsetWrite.RelativeTo)
 			}
-			// offset is written as a 4-byte pointer value at offset-write address
-			if ps.OffsetWrite.Offset > (other.Size - SizeLBA48Pointer) {
+			// make sure there is enough room left in the structure to write the pointer
```

Comment:
So we actually write the pointer into the disk? I didn't anticipate that. Let's catch up tomorrow :)